platform_disk_api/api.py

```diff
@@ -141,10 +158,21 @@ def _get_disk_uri(self, disk: Disk) -> str:
         return f"{base}/{disk.owner}/{disk.id}"
 
     def _get_disk_read_perm(self, disk: Disk) -> Permission:
-        return Permission(self._get_disk_uri(disk), "read")
+        if disk.project_name == disk.owner:
+            return Permission(self._get_disk_uri(disk), "read")
+        return self._get_project_disks_read_perm(disk.project_name, disk.org_name)
 
     def _get_disk_write_perm(self, disk: Disk) -> Permission:
-        return Permission(self._get_disk_uri(disk), "write")
+        if disk.project_name == disk.owner:
+            return Permission(self._get_disk_uri(disk), "write")
+        return self._get_project_disks_write_perm(disk.project_name, disk.org_name)
```

Comment:
if we add the disk id into the project_disks URI, the auth service will still check permissions according to hierarchy. so I guess this could further be simplified to a one-liner. maybe we could just patch `_get_disk_uri` to account for `project_name`? though I might be missing something.